src/App.tsx

```diff
@@ -17,18 +58,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedOption={selectedOption}
+                searchQuery={searchQuery}
+                onChangeSelect={setSelectedOption}
+                onChangeSearch={setSearchQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                todos.length === 0
+                  ? <Loader />
+                  : (
+                    <TodoList
+                      todos={filteredTodos}
+                      selectedTodoId={todoId}
+                      selectTodo={handleSelectTodo}
+                    />
+                  )
+              }
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal todo={selectedTodo} setTodoId={setTodoId} />
```

Comment:
i think it will be more informative

```suggestion
        <TodoModal todo={selectedTodo} onClose={() => setTodoId(0)} />
```